pkg/api/customization/logging/list_handler.go

```diff
@@ -86,47 +93,58 @@ func convertToClusterResource(schema *types.Schema, clusterLogging v3.ClusterLog
 }
 
 func (h *ClusterLoggingHandler) setEmbeddedEndpoint(cl *v3.ClusterLogging) error {
-	if cl.Spec.EmbeddedConfig != nil {
-		esLabels := fmt.Sprintf("%s=%s", loggingconfig.LabelK8sApp, loggingconfig.EmbeddedESName)
-		espods, err := h.CoreV1.Pods(loggingconfig.LoggingNamespace).List(metav1.ListOptions{LabelSelector: esLabels})
-		if err != nil {
-			return err
-		}
-		esservice, err := h.CoreV1.Services(loggingconfig.LoggingNamespace).Get(loggingconfig.EmbeddedESName, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		if len(esservice.Spec.Ports) == 0 {
-			return fmt.Errorf("get service %s node port failed", loggingconfig.EmbeddedKibanaName)
-		}
-		var esPort int32
-		for _, v := range esservice.Spec.Ports {
-			if v.Name == "http" {
-				esPort = v.NodePort
-				break
-			}
+	esLabels := fmt.Sprintf("%s=%s", loggingconfig.LabelK8sApp, loggingconfig.EmbeddedESName)
+	espods, err := h.CoreV1.Pods(loggingconfig.LoggingNamespace).List(metav1.ListOptions{LabelSelector: esLabels})
```

Comment:
mm, why are we not using lister for listing the pods? We should read them from cache. And lets get rid of CoreV1. Just use the fields instead, listers for read, interfaces for update.